exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/mergereceiver/MergingRecordBatch.java

```diff
@@ -795,6 +788,8 @@ private void generateComparisons(final ClassGenerator<?> g, final VectorAccessib
    * @param node Reference to the next record to copy from the incoming batches
    */
   private boolean copyRecordToOutgoingBatch(final Node node) {
+    assert outgoingPosition < OUTGOING_BATCH_SIZE
```

Comment:
Doing this assert check on each record would add some overhead.  Since the caller already is checking outgoingBatchHasSpace in the while loop, this check is not needed.  You could add a javadoc comment that caller is expected to do the validity check.

Reply:
I added the assert to avoid possible errors during further code refactoring. As it is an assert that will not affect performance in production and there is another assert already, I'd prefer to keep it.

Reply:
Ok.  Updated changes lgtm.